mne_connectivity/decoding/decomposition.py

```diff
@@ -465,3 +469,306 @@ def get_transformed_indices(self):
+            vlim,
+            cnorm,
+            colorbar,
+            cbar_fmt,
+            units,
+            axes,
+            name_format,
+            nrows,
+            ncols,
+            show,
+        )
+
+    def _plot_filters_patterns(
+        self,
+        plot_data,
+        info,
+        components,
+        ch_type,
+        scalings,
+        sensors,
+        show_names,
+        mask,
+        mask_params,
+        contours,
+        outlines,
+        sphere,
+        image_interp,
+        extrapolate,
+        border,
+        res,
+        size,
+        cmap,
+        vlim,
+        cnorm,
+        colorbar,
+        cbar_fmt,
+        units,
+        axes,
+        name_format,
+        nrows,
+        ncols,
+        show,
+    ):
+        """Plot filters/targets for components."""
+        # Sort inputs
+        _validate_type(info, Info, "`info`", "mne.Info")
+        if components is None:
+            components = np.arange(self.n_components)
+
+        # plot seeds and targets
+        figs = []
+        for group_idx, group_name in zip([0, 1], ["Seeds", "Targets"]):
+            # create info for seeds/targets
+            group_info = deepcopy(info)
+            group_info = pick_info(group_info, self.indices[group_idx], copy=False)
+            with group_info._unlock():
+                group_info["sfreq"] = 1.0  # 1 component per time point
+            # create Evoked object
+            evoked = EvokedArray(plot_data[group_idx], group_info, tmin=0)
+            # then call plot_topomap
+            figs.append(
+                evoked.plot_topomap(
+                    times=components,
+                    average=None,  # do not average across independent components
+                    ch_type=ch_type,
+                    scalings=scalings,
+                    sensors=sensors,
+                    show_names=show_names,
+                    mask=mask,
+                    mask_params=mask_params,
+                    contours=contours,
+                    outlines=outlines,
+                    sphere=sphere,
+                    image_interp=image_interp,
+                    extrapolate=extrapolate,
+                    border=border,
+                    res=res,
+                    size=size,
+                    cmap=cmap,
+                    vlim=vlim,
+                    cnorm=cnorm,
+                    colorbar=colorbar,
+                    cbar_fmt=cbar_fmt,
+                    units=units,
+                    axes=axes,
+                    time_format=f"{self._conn_estimator.name}%01d"
+                    if name_format is None
+                    else name_format,
+                    nrows=nrows,
+                    ncols=ncols,
+                    show=False,  # set Seeds/Targets suptitle first
+                )
+            )
+            figs[-1].suptitle(group_name)  # differentiate seeds from targets
+            if show:
+                figs[-1].show()
```

Comment:
You should use `plt_show(fig=figs[-1])` from `mne.utils` as it does some stuff with warnings IIRC and also deals with the conditional
```suggestion
            plt_show(show, fig=figs[-1])
```

Reply:
Done!